### DISCUSSION TITLE
reset atoms if Provider initialValues changes

### DISCUSSION THREAD

**Discussion by horaciosystem** [pmndrs/jotai#1462] [Q&A] [answered]
I'm using a Provider to initialize an atom value (**userAtom**), where the initial value (**user**) comes from a async call up in the component tree.
```js
<Provider
  initialValues={[
  	[userAtom, user]
  ]}
  <Component />
</Provider>
```
In this case the **userAtom** gets its value from **user** on the very first render, so as the **user** is yet being fetched, it will get a default value, like null, undefined, and not the desired value after user loads.
I'm looking for an alternative for this, something like a prop to make the Provider reinitialize the atom values if the initial values change, pretty much what a some form libraries allows.

From formik docs:
> enableReinitialize?: boolean
> Default is false. Control whether Formik should reset the form if initialValues changes (using deep equality).

The output could be something like this:
```js
<Provider
  enableReinitialize //Name taken from formik
  initialValues={[
  	[userAtom, user]
  ]}
  <Component />
</Provider>
```

1. Does it make sense?

2. If not, what alternative I could use to make the Provider/atoms reactive? I'm looking for an alternative without going for a Jotai libraries integration path. Many legacy extensive code bases can adopt Jotai incrementally without going for a full refactoring.

**Comment by dai-shi** [comment, marked as answer]
In this use case, our current suggestion is to remount the Provider component. 

```jsx
<Provider
  initialValues={[[userAtom, user]]}
  key={user.id}
>
```

Hope it works.
